From ed35213c8f33dc49ed1dcea9da37226fbbe8d40e Mon Sep 17 00:00:00 2001
From: Mathew McBride <matt@traverse.com.au>
Date: Mon, 22 Jul 2024 19:32:44 +1000
Subject: [PATCH 1/4] wifi_hal: do not create any vaps other than
 private_ssid_* for now

The mt76 driver appears to have issues accepting parameters for the various
non private_ssid (iot/hotspot/mesh/etc.), causing none of the 5G SSIDs to
come up.

Until this issue is resolved, only create the private_ssid_2g/5g.

Signed-off-by: Mathew McBride <matt@traverse.com.au>
---
 src/wifi_hal.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/wifi_hal.c b/src/wifi_hal.c
index b2ffbe5..95eca2f 100644
--- a/src/wifi_hal.c
+++ b/src/wifi_hal.c
@@ -1352,6 +1352,11 @@ INT wifi_hal_createVAP(wifi_radio_index_t index, wifi_vap_info_map_t *map)
     for (i = 0; i < map->num_vaps; i++) {
         vap = &map->vap_array[i];
 
+        if (vap->vap_index != 0 && vap->vap_index != 1) {
+            wifi_hal_info_print("%s:%d: Ignoring VAP index %d\n", __func__, __LINE__, vap->vap_index);
+            continue;
+        }
+
         wifi_hal_info_print("%s:%d: vap index:%d vap_name = %s create vap\n", __func__, __LINE__,
             vap->vap_index, vap->vap_name);
 
-- 
2.39.5

